Frontend/ui-driver/src/Screens/DriverCompleteProfileScreen/Controller.purs

```diff
@@ -0,0 +1,219 @@
+            | ProfileDataAPIResponseAction DriverProfileDataRes
+
+instance showAction :: Show Action where
+  show _ = ""
+instance loggableAction :: Loggable Action where
+  performLog action appId = case action of
+    _ -> pure unit
+
+data ScreenOutput = SubmitRes DriverCompleteProfileScreenState | GoToProfile DriverCompleteProfileScreenState
+
+eval :: Action -> DriverCompleteProfileScreenState -> Eval Action ScreenOutput DriverCompleteProfileScreenState
+
+eval (OnClickPledge title isSelected) state = do
+  if title == "Other" then do
+    if isSelected then continue state { data { inputTextState { others { pledge = "" } } } } else continue state { data { inputTextState { component = Pledge } }, props { showInputTextView = true } }
+  else continue state {data {pledge = if isSelected == true then DA.filter (\item -> (item /= title) ) state.data.pledge else state.data.pledge <> [title] }}
+
+eval (ProfileDataAPIResponseAction res) state = do 
+  let DriverProfileDataRes resp = res 
+      _ = spy "res is" res
+      pledgeOthers = (DA.filter (\item -> (item /= "Safe Journey" && item /= "Clean Car" && item /= "On-time Pick up" && item /= "Regular Maintenance") )) resp.pledges
+      whyNyOthers = spy "whyNyOthers" $ (DA.filter (\item -> (item /= "Buy New Home" && item /= "Kid's Education" && item /= "Buy New Vehicle") )) resp.aspirations
+  continue state{data{
+    pledge = (DA.filter (\item -> (item == "Safe Journey" || item == "Clean Car" || item == "On-time Pick up" || item == "Regular Maintenance") )) resp.pledges, 
+    whyNy = (DA.filter (\item -> (item == "Buy New Home" || item == "Kid's Education" || item == "Buy New Vehicle") )) resp.aspirations, 
+    drivingSince = resp.drivingSince, 
+    vehicalOffer = resp.vehicleTags, 
+    inputTextState { others {pledge = if DA.length pledgeOthers == 0 then "" else fromMaybe "" $ DA.head pledgeOthers, whyNy = if DA.length whyNyOthers == 0 then "" else fromMaybe "" $ DA.head whyNyOthers}},
+    homeTown = resp.hometown, 
+    datePickerState {activeIndex = if resp.drivingSince == Nothing then 0 else 69 - ((fromMaybe 0 $ fromString (EHC.convertUTCtoISC (EHC.getCurrentUTC "") "YYYY")) - fromMaybe 0 resp.drivingSince)},
+    addImagesState{ images = DA.mapWithIndex(\idx item -> {image : item, imageName :( "Image" <> show idx)})resp.otherImages, imageMediaIds = resp.otherImageIds, stateChanged = true}, 
+    uploadedImagesIds = resp.otherImageIds, addedImages = DA.mapWithIndex(\idx item -> {image : item, imageName : ("Image" <> show idx)})resp.otherImages}}
+
+eval (OnClickVehicalOffer title isSelected) state = continue state {data {vehicalOffer = if isSelected == true then DA.filter (\item -> (item /= title) ) state.data.vehicalOffer else state.data.vehicalOffer <> [title] }}
+
+eval (OnClickLanguages title isSelected) state = continue state {data {languages = if isSelected == true then DA.filter (\item -> (item /= title) ) state.data.languages else state.data.languages <> [title] }}
```

Comment:
as isSelected is already boolean , no need of isSelected == true

there are multiple places , fix all